dhcpv6/async.go

```diff
@@ -0,0 +1,214 @@
+package dhcpv6
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"sync"
+	"time"
+)
+
+// ClientAsync implements an asynchronous DHCPv6 client
+type ClientAsync struct {
+	ReadTimeout  time.Duration
+	WriteTimeout time.Duration
+	LocalAddr    net.Addr
+	RemoteAddr   net.Addr
```

Comment:
these four fields above are also all of the fields in `Client`. What about embedding it?

Reply:
This might be confusing as it will give the access to synchronous methods on an async client

Reply:
you're absolutely right, I didn't think of that